### ISSUE TITLE
Exposing Routes on Router

### ISSUE THREAD

**Issue by pquerna** [gocraft/web#40]
Hi,

I was looking at adding the ability to advertise `web` routes to Vulcan in a package outside of gocraft/web.  This would be kinda similar to how Mailgun's scroll registers its routes into etcd, so then vulcand can send it traffic:

https://github.com/mailgun/scroll

However the Routes are currently all private, and there is no way to get them once they have been added.   This could be as easy as adding a `Routes() []*Routes` or similar function, so then it is possible to introspect the routes on a router.

**Comment by cypriss** [comment]
@pquerna Sorry for the delay in responding to this. Is this still something you're interested in?

**Comment by pquerna** [comment]
@cypriss Yes.  Right now we override all of the Get/Put/Post functions in our own structure to capture the routes as they are being added.
